substra/sdk/backends/local/compute/spawner.py

```diff
@@ -36,14 +36,16 @@ def _unzip(archive, to_):
 
 def _uncompress(archive, to_):
     """Uncompress tar or zip archive to destination."""
-    if archive.endswith('.zip'):
-        handler = _unzip
-    elif archive.endswith('.tgz') or archive.endswith('.tar.gz'):
-        handler = _untar
-    else:
-        raise exceptions.InvalidRequest(f"Cannot uncompres '{archive}'", 400)
-
-    handler(archive, to_)
+    success = False
+    errors = list()
+    for handler in [_unzip, _untar]:
+        try:
+            handler(archive, to_)
+            success = True
+        except Exception as e:
+            errors.append(e)
+    if not success:
+        raise exceptions.InvalidRequest(f"Cannot uncompress '{archive}', error: \n{errors}", 400)
```

Comment:
What's the purpose of modifying this code?

The main drawback of the new code is that we will always run both handlers.

One enhancement could be to catch 2 different kind of errors:
- "Invalid archive extensions"
- "Cannot uncompress"
?

Reply:
Yes, if we could trust the extension (.zip or .tar.gz) of the archive to know which handler to use it would be better. The problem is that when we download the archive from the remote, we rename it and lose the extension, so we have to try both handlers.

Catching their specific errors is a good idea !

Reply:
After a quick verification, the tarfile and zipfile packages provide `is_tarfile` and `is_zipfile` functions, which is better than the `try...except...` that I used before !

Reply:
Looks great!